FakebookNotifications/FakebookNotifications.Testing/IntegrationTests/NotificationControllerTest.cs

```diff
@@ -0,0 +1,188 @@
+﻿using FakebookNotification.WebApi.Controllers;
+using FakebookNotifications.DataAccess.Models;
+using FakebookNotifications.DataAccess.Repositories;
+using FakebookNotifications.WebApi.Hubs;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace FakebookNotifications.Testing.IntegrationTests
+{
+    public class NotificationControllerTest
+    {
+
+        private static Domain.Models.Notification GetDummyNotification(string type)
+        {
+            return new()
+            {
+                Id = "0123",
+                Type = new(type, 1),
+                LoggedInUserId = "test@test.com",
+                TriggerUserId = "test@test.com",
+                HasBeenRead = false,
+                Date = new DateTime(2021, 04, 05)
+            };
+        }
+
+        private readonly Domain.Models.User testUser1 = new()
+        {
+            Id = "01",
+            Connections = new List<string>{
+                "00", "02", "03"
+            },
+            Email = "test@test.com"
+        };
+
+        private readonly List<string> groupIds = new()
+        {
+            "test@test.com",
+            "group1",
+            "group2",
+            "group3"
+        };
+
+        private readonly List<string> clientIds = new() { "00", "01", "02", "03", "04", "05" };
+
+        private readonly Mock<IHubCallerClients> mockClients = new();
+        private readonly Mock<IGroupManager> mockGroups = new();
+        private readonly Mock<IClientProxy> mockClientProxy = new();
+        private readonly Mock<HubCallerContext> mockContext = new();
+        private readonly NotificationHub mockHub;
+
+        private readonly UserRepo _userRepo;
+        private readonly NotificationsRepo _noteRepo;
+        private readonly NotificationsDatabaseSettings settings;
+        private readonly NullLogger<NotificationsContext> _logger;
```

Comment:
use a consistent naming convention for private fields - in this project, that's an initial underscore